lib/phase/functional.js

```diff
@@ -254,12 +254,29 @@ module.exports = async ({
     const doc = flattenedDoc;
     let errors = [];
     let warnings = [];
+    let pipelineOnly = true;
 
     // Jobs
     errors = errors.concat(validateJobSchema(doc));
 
+    if (doc.stages) {
+        pipelineOnly = false;
+    }
+
     // Workflow graph
-    doc.workflowGraph = await generateWorkflowGraph(doc, triggerFactory, pipelineId);
+    const workflowGraphs = await generateWorkflowGraph(doc, triggerFactory, pipelineId, pipelineOnly);
```

Comment:
We should always keep the type and structure of the return value for a function irrespective of the arguments being passed.
```suggestion
    const {pipelineWorkflow, stageWorkflows} = await generateWorkflowGraph(doc, triggerFactory, pipelineId, pipelineOnly);
```